builders/interact/ethereum-api/dev-env/hardhat.md

```diff
@@ -0,0 +1,247 @@
+
+```sh
+touch Box.sol
+```
+
+Open the file and add the following contract to it:
+
+```solidity
+// contracts/Box.sol
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.1;
+
+contract Box {
+    uint256 private value;
+
+    // Emitted when the stored value changes
+    event ValueChanged(uint256 newValue);
+
+    // Stores a new value in the contract
+    function store(uint256 newValue) public {
+        value = newValue;
+        emit ValueChanged(newValue);
+    }
+
+    // Reads the last stored value
+    function retrieve() public view returns (uint256) {
+        return value;
+    }
+}
+```
+
+## Hardhat Configuration File {: #hardhat-configuration-file }
+
+Before you can deploy the contract to your ContainerChain, you'll need to modify the Hardhat configuration file and create a secure file to store your private key in.
+
+You can modify the `hardhat.config.js` file to use either the Tanssi Dancebox Demo EVM ContainerChain or your own ContainerChain:
+
+
+```js
+// 1. Import the Ethers plugin required to interact with the contract
+require('@nomicfoundation/hardhat-ethers');
+
+// 2. Add your private key that is funded with tokens of your ContainerChain
+const privateKey = 'INSERT_PRIVATE_KEY';
+
+module.exports = {
+  // 3. Specify the Solidity version
+  solidity: '0.8.1',
+  networks: {
+    // 4. Add the network specification for your Tanssi EVM ContainerChain
+    dancebox: {
+      url: '{{ networks.dancebox.rpc_url }}',
+      chainId: {{ networks.dancebox.chain_id }}, // Fill in the EVM ChainID for your ContainerChain
+      accounts: [privateKey]
+    }
+  }
+};
+```
+
+Congratulations! You are now ready for deployment!
+
+## Compiling Solidity {: #compiling-solidity }
+
+To compile the contract you can simply run:
+
+```sh
+npx hardhat compile
+```
+
+![Hardhat Contract Compile](/images/builders/interact/ethereum-api/dev-environments/hardhat/hardhat-2.png)
+
+After compilation, an `artifacts` directory is created: it holds the bytecode and metadata of the contract, which are `.json` files. It’s a good idea to add this directory to your `.gitignore`.
+
+## Deploying the Contract {: #deploying-the-contract }
+
+In order to deploy the `Box.sol` smart contract, you will need to write a simple deployment script. You can create a new directory for the script and name it `scripts` and add a new file to it called `deploy.js`:
+
+```sh
+mkdir scripts && cd scripts
+touch deploy.js
+```
+
+Next, you need to write your deployment script which can be done using `ethers`. Because you'll be running it with Hardhat, you don't need to import any libraries.
+
+To get started, take the following steps:
+
+1. Create a local instance of the contract with the `getContractFactory` method
+2. Use the `deploy` method that exists within this instance to instantiate the smart contract
+3. Wait for the deployment by using `deployed`
+4. Once deployed, you can fetch the address of the contract using the contract instance.
+
+```js
```

Comment:
please run prettier on this snippet, looks like it's using 4 spaces instead of 2